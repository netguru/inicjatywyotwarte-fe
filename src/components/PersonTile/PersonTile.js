import React from 'react'
import styled from 'styled-components'

const Container = styled.div`
  width: 100%;
  height: calc(${props => props.theme.personTileWidth} + 120px);
  font-size: 0.85rem;
`
Container.displayName = 'PersonTileContainer';

const ImageContainer = styled.div`
  width: 100%;
  height: ${props => props.theme.persontil};

  img {
    border-radius: ${props => props.theme.borderRadiusMedium};
    max-width: 100%;
    max-height: 100%;
  }
`

const Name = styled.h3`
  margin: 0.6rem 0 0.5rem 0;
  font-weight: 500;
`
Name.displayName = 'Name';

const Role = styled.div`
  margin: 0.4rem 0;
  font-weight: 300;
  line-height: 1rem;
`
Role.displayName = 'Role';

const Company = styled.div`
  margin: 0.8rem 0;
  color: ${props => props.theme.fontColorMedium};
  font-weight: 300;
`
Company.displayName = 'Company';

const PersonTile = ({ photo, name, role, company }) => {
  return (
    <Container>
      <ImageContainer>
        <img src={photo} alt='' />
      </ImageContainer>
      <Name>{name}</Name>
      <Role>{role}</Role>
      <Company>{company}</Company>
    </Container>
  )
}

export default PersonTile
