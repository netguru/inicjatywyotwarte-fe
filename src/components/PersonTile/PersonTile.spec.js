import React from 'react'
import PersonTile from './PersonTile'
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('PersonTile', () => {
  test('renders person tile container', () => {
    const wrapper = shallow(<PersonTile/>);
    expect(wrapper.find('PersonTileContainer').length).toEqual(1);
  });

  test('renders proper attributes', () => {
    const wrapper = shallow(
      <PersonTile
        name={"person name"}
        role={"person role"}
        company={"person company"}
      />
      );

    expect(wrapper.find('Name').text()).toEqual('person name');
    expect(wrapper.find('Role').text()).toEqual('person role');
    expect(wrapper.find('Company').text()).toEqual('person company');
  });

  test('match snapshot', () => {
    const wrapper = shallow(<PersonTile/>);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});


