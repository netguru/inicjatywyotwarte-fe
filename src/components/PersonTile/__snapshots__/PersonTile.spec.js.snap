// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonTile match snapshot 1`] = `
<PersonTileContainer>
  <styled.div>
    <img
      alt=""
    />
  </styled.div>
  <Name />
  <Role />
  <Company />
</PersonTileContainer>
`;
